internal/pkg/instrumentation/probe/interfaces.go

```diff
@@ -0,0 +1,73 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package probe provides instrumentation probe types and definitions.
+package probe
+
+import (
+	"log/slog"
+)
+
+// BaseProbe is the most basic type of Probe. It must support configuration,
+// loading and attaching eBPF programs, running, and closing.
+// All Probes must implement this Base functionality.
+type BaseProbe interface {
+	// ID is a unique identifier for the Probe.
+	// All types of Probes must provide a unique identifier for the framework.
+	ID() ID
+
+	// GetLogger returns an *slog.Logger for this Probe.
+	GetLogger() *slog.Logger
+
+	// Load loads the eBPF programs and maps required by the Probe into memory.
+	// The specific types of programs and maps are implemented by the Probe.
+	Load() error
+
+	// Attach attaches the eBPF programs to trigger points in the process.
+	// The specific attachment points are implemented by the Probe.
+	Attach() error
+
+	// ApplyConfig updates the Probe's current Config with the provided Config
+	// interface. It is up to the Probe to implement type conversion to any custom
+	// config formats it defines, to support options specific to the Probe.
+	ApplyConfig(Config) error
+}
+
+// RunnableProbe is a Probe that Runs.
+type RunnableProbe interface {
+	BaseProbe
+
+	// Run starts the Probe.
+	Run()
```

Comment:
Do we want to pass `context` here?
Do we want to commit to the current approach that `Run` is blocking until canceled?